packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -539,60 +564,63 @@ export class TokenDetectionController extends StaticIntervalPollingController<
         tokensToDetect.push(tokenAddress);
       }
     }
+
     const slicesOfTokensToDetect = [];
     slicesOfTokensToDetect[0] = tokensToDetect.slice(0, 1000);
     slicesOfTokensToDetect[1] = tokensToDetect.slice(
       1000,
       tokensToDetect.length - 1,
     );
```

Comment:
In case you want to fix https://github.com/MetaMask/core/issues/3661 while we're here:

```suggestion
    const sliceOfTokensToDetect = [];
    for (let i = 0, size = 1000; i < tokensToDetect.length; i += size) {
      sliceOfTokensToDetect.push(tokensToDetect.slice(i, i + size));
    }
```

Reply:
Looks good! Applied here: https://github.com/MetaMask/core/pull/3938/commits/f5c3194a83ea8394194bf30f95a7c72f3b94c389